### PR TITLE
use output of executable stubs file

### PR DESCRIPTION
**Pull request by JamesClonk**
Signed-off by Fabio Berchtold <fabio.berchtold@swisscom.com>

### CODE REVIEW

**Comment by JamesClonk** [comment]
This will allow to have executable stubs in the stubs folder to run and use their output for merging.
Such a script could potentially read information from anywhere like a database, other manifests files, etc.

For example one could have an executable **stubs/diego.yml** which outputs a `spiff merge ../.releases/diego-release/config-from-cf.yml ../.releases/diego-release/config-from-cf-internal.yml ../.deployments/cf.yml`, so it would not be needed to maintain any values from cf.yml also in the diego.yml manifest.

File: lib/bosh/workspace/project_deployment.rb [hunk 1]
```diff
@@ -44,7 +44,13 @@ def manifest
     def stub
       return @stub unless @stub.nil?
       stub_file = File.expand_path(File.join(file_dirname, "../stubs", file_basename))
-      @stub = File.exist?(stub_file) ? Psych.load(File.read(stub_file)) : {}
+      @stub = File.exist?(stub_file) ?
+        Psych.load(File.executable?(stub_file) ? execute_stub(stub_file) : File.read(stub_file))
+      : {}
```
Comment (rkoster) [inline lib/bosh/workspace/project_deployment.rb]: Would be nice to extract some of the logic here into separate methods, and don't use single line if statements when doing multiline

**Comment by rkoster** [comment]
Code review done, when the above is fixed it's ready to be merged.

**Comment by JamesClonk** [comment]
@rkoster ok, done :smile:

**Pull request merged**